### PR TITLE
Cv2 3700 sscd image similarity algorithm

### PR DESCRIPTION
**Pull request by ahmednasserswe**
## Description
Implementing SSCD-copy-detection algorithm. 

Reference: Cv2 3700

## How has this been tested?
Tested locally, to be tested in the PR by automated and integration tests

## Have you considered secure coding practices when writing this code?
no security concerns found

### CODE REVIEW

**Review by computermacgyver — CHANGES_REQUESTED** [review]
Please pull in the latest from the `develop` branch @ahmednasserswe and re-test

File: app/main/model/image.py [hunk 1]
```diff
@@ -52,11 +54,12 @@ def from_url(url, doc_id, context={}, created_at=None):
     raw = remote_response.read()
     im = Image.open(io.BytesIO(raw)).convert('RGB')
     phash = compute_phash_int(im)
+    sscd = None
```
Comment (DGaffney) [inline app/main/model/image.py:57]: I'm not seeing where, if ever, sscd gets set - am I just missing it?

**Review by DGaffney — CHANGES_REQUESTED** [review]
I'm a bit confused as to how SSCD is set - if this is just establishing it as a field then this is fine, otherwise this needs more work.

File: app/main/model/image.py (hunk 1 above)
Comment (computermacgyver) [reply app/main/model/image.py:57]: This PR creates the field (nullable) and always set it to null. The next PR should actually set it. My thinking was to break it into two PRs with this first one being "safe" insofar as it just creates the DB column (and always sets it to null)

**Review by DGaffney — APPROVED** [review]

**Review by computermacgyver — APPROVED** [review]

**Comment by ahmednasserswe** [comment]
Implementing SSCD-copy-detection algorithm.
Adding a nullable column to image of type pgvector to contain the sscd embedding vectors.

Reference: Cv2 3700

**Pull request merged**